core/math/geometry.h

```diff
@@ -785,6 +789,165 @@ class Geometry {
 	}
 
+	static Vector<Vector<Point2> > _clipper_execute(ClipperLib::ClipType op, const Vector<Point2> &p_polypath_a, const Vector<Point2> &p_polypath_b, bool is_a_open = false) {
+		using namespace ClipperLib;
+
+		Path path_a, path_b;
+
+		// Need to scale points (Clipper's requirement for robust computation)
+		for (int i = 0; i != p_polypath_a.size(); ++i) {
+			path_a << IntPoint(p_polypath_a[i].x * SCALE_PRECISION, p_polypath_a[i].y * SCALE_PRECISION);
+		}
+		for (int i = 0; i != p_polypath_b.size(); ++i) {
+			path_b << IntPoint(p_polypath_b[i].x * SCALE_PRECISION, p_polypath_b[i].y * SCALE_PRECISION);
+		}
+		Clipper clp;
+		clp.AddPath(path_a, ptSubject, !is_a_open); // forward compatible with Clipper 10.0.0
+		clp.AddPath(path_b, ptClip, true); // polylines cannot be set as clip
+
+		Paths paths;
+
+		if (is_a_open) {
+			PolyTree tree; // needed to populate polylines
+			clp.Execute(op, tree);
+			OpenPathsFromPolyTree(tree, paths);
+		} else {
+			clp.Execute(op, paths); // works on closed polygons only
+		}
+		// Have to scale points down now
+		Vector<Vector<Point2> > polypaths;
+
+		for (Paths::size_type i = 0; i < paths.size(); ++i) {
+			Vector<Vector2> polypath;
+
+			const Path &path = paths[i];
+
+			for (Paths::size_type j = 0; j != path.size(); ++j) {
+				polypath.push_back(Point2(
+						static_cast<real_t>(path[j].X) / SCALE_PRECISION,
+						static_cast<real_t>(path[j].Y) / SCALE_PRECISION));
+			}
+			polypaths.push_back(polypath);
+		}
+		return polypaths;
+	}
+
+	enum PolyJoinType {
+		JOIN_SQUARE,
+		JOIN_ROUND,
+		JOIN_MITER
+	};
+	enum PolyEndType {
+		END_POLYGON,
+		END_JOINED,
+		END_BUTT,
+		END_SQUARE,
+		END_ROUND
+	};
+
+	static Vector<Vector<Point2> > _clipper_offset_execute(const Vector<Point2> &p_polypath, real_t p_delta, PolyJoinType p_join_type, PolyEndType p_end_type) {
+		using namespace ClipperLib;
+
+		JoinType jt;
+
+		switch (p_join_type) {
+			case JOIN_SQUARE: jt = JoinType::jtSquare; break;
+			case JOIN_ROUND: jt = JoinType::jtRound; break;
+			case JOIN_MITER: jt = JoinType::jtMiter; break;
+		}
+
+		EndType et;
+
+		switch (p_end_type) {
+			case END_POLYGON: et = EndType::etClosedPolygon; break;
+			case END_JOINED: et = EndType::etClosedLine; break;
+			case END_BUTT: et = EndType::etOpenButt; break;
+			case END_SQUARE: et = EndType::etOpenSquare; break;
+			case END_ROUND: et = EndType::etOpenRound; break;
+		}
+		ClipperOffset co;
+		Path path;
+
+		// Need to scale points (Clipper's requirement for robust computation)
+		for (int i = 0; i != p_polypath.size(); ++i) {
+			path << IntPoint(p_polypath[i].x * SCALE_PRECISION, p_polypath[i].y * SCALE_PRECISION);
+		}
+		co.AddPath(path, jt, et);
+
+		Paths paths;
+		co.Execute(paths, p_delta * SCALE_PRECISION); // inflate/deflate
+
+		// Have to scale points down now
+		Vector<Vector<Point2> > polypaths;
+
+		for (Paths::size_type i = 0; i < paths.size(); ++i) {
+			Vector<Vector2> polypath;
+
+			const Path &path = paths[i];
```

Comment:
This shadows the variable in line 869, which makes it fail the Travis build.

Reply:
Shall be fixed, yet fixing some other warnings results in warnings in other Travis jobs. 😄